Grandeur results for the Finished Tab

erinyoung opened the pull request:
My apologies for taking so long for this!!!

This takes the results from Grandeur and formats it for the ARLN and FINISHED tabs on our google sheet.

This will be deprecated once LW8 is live, but... that's going to take awhile.

erinyoung committed:
adding script

erinyoung committed:
added usage for grandeur_to_sheets.py

erinyoung commented:
This is what gets printed to the screen, so I'm thinking that I'm likely repeating steps that don't need to be repeated

```
24-Nov-25 09:33:41 - Getting samples from reads/SampleSheet.csv
24-Nov-25 09:33:41 - Extracting information from 1234/definition/grandeur/grandeur_summary.tsv
24-Nov-25 09:33:41 - Double checking Escherichia organism with shigatyper results
24-Nov-25 09:33:41 - Getting kraken2 results
24-Nov-25 09:33:41 - Getting blobtools results
24-Nov-25 09:33:41 - Getting information from 1234/definition/grandeur/fastani/
24-Nov-25 09:33:43 - Getting information from 1234/definition/grandeur/mash/
24-Nov-25 09:33:44 - Getting salmonella serotype information from 1234/definition/grandeur/seqsero2/
24-Nov-25 09:33:44 - Getting information from 1234/definition/grandeur/mlst/
24-Nov-25 09:33:46 - Getting information from 1234/definition/grandeur/emmtyper/
24-Nov-25 09:33:46 - Getting information from 1234/definition/grandeur/ncbi-AMRFinderplus/
24-Nov-25 09:33:49 - Getting information from 1234/definition/grandeur/ncbi-AMRFinderplus/
24-Nov-25 09:33:49 - Getting information from 1234/definition/grandeur/ncbi-AMRFinderplus/
24-Nov-25 09:33:50 - Getting information from 1234/definition/grandeur/ncbi-AMRFinderplus/
24-Nov-25 09:33:50 - Writing file for Finished tab
24-Nov-25 09:33:50 - Created finished_tab.{txt,tsv}
24-Nov-25 09:33:50 - Writing file for ARLN tab
24-Nov-25 09:33:50 - Created arln_tab.{txt,tsv}
```

erinyoung committed:
ensuring type

erinyoung committed:
updated for miseq i100 samplesheets

erinyoung commented:
I updated this to account for the new MiSeq i100 sample sheets.

It works as "normal", but is a bit more flexible now.

erinyoung committed:
added str types for sample names that are just numbers

erinyoung committed:
adding in clarity_query_function

erinyoung committed:
updated usage for grandeur_to_results to include run

erinyoung committed:
fixed sample sheet input function

jwarnn reviewed (approved):
Looks good. Ran as expected.

jwarnn commented on grandeur_to_sheets.py:
```diff
@@ -0,0 +1,711 @@
+    if 'ipaH' in amr:
+        genus_matches = matches[matches['organism'].str.contains('Shigella', case=False, na=False)]
+        organism = f"{genus_matches.sort_values(by='ANI estimate', ascending=False).iloc[0]['organism']} (ipaH+)"
+    else:
```
This line gave me an issue when the first time I tested the script it worked fine. For run UT-SH00354-250617 it errors out on this else block. ```genus_matches``` returns empty so the iloc command on the next line errors as there is nothing to lock on. Not sure what the fix is here but the offending sample was a ```Shigella_sonnei``` with no iphH in the ```amrfinder_genes_(per_cov/per_ident)``` column.

erinyoung committed:
fixing fastani df and serotypefinder results

erinyoung commented:
I think I've fixed this error. 

1. I didn't realize I wasn't passing the entire fastani dataframe to this function, that's been resovled
2. This is also where the serotypefinder results should be determined, and that is now fixed

Sorry!

jwarnn commented:
I ran it on the last two runs and I got it too work.

erinyoung committed:
check ipaH in virulence genes

erinyoung committed:
allows for runs with no E. coli

erinyoung committed:
fixed sample id for files

erinyoung commented:
@jwarnn , I think I fixed the sample naming issue. Do you think this is good to merge in?

jwarnn commented:
I tested in on 6 runs; UT-SH00354-250529, UT-SH00354-250527, UT-SH00281-250528, UT-SH00354-250625, UT-SH00792-250630, UT-SH00354-250702. All of them had completely filled in finished_tab.tsv files. All of the Ecoli and Shigella samples looked to be named correctly depending on ipaH- and the Sero types looked to be formatted correctly as far as I understand. I think this can be merged.
